### PR TITLE
OpenVPN and Generic VPN switch module

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Added OpenVPN and GenericVPN switch module

# Impacts
RADIUS VPN flow

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* OpenVPN and GenericVPN Switch module

### CODE REVIEW

File: lib/pf/Switch/OpenVPN.pm [hunk 1]
```diff
@@ -0,0 +1,163 @@
+package pf::Switch::OpenVPN;
```
Comment (jrouzierinverse) [inline lib/pf/Switch/OpenVPN.pm:1]:
```
#   Failed test 'no tab character in /usr/local/pf/lib/pf/Switch/OpenVPN.pm'
#   at coding-style.t line 49.
# Looks like you failed 1 test of 1051.
coding-style.t .....................................................
```

File: docs/network/vpn_configuration.asciidoc [hunk 2]
```diff
@@ -100,3 +100,118 @@ Before trying to configure PacketFence with the Cisco ASA first be sure that whe
...
+ verb 3
+ verify-client-cert none
+
+Next you need to compile the radius extention for openvpn:
+
+ wget https://github.com/ValdikSS/openvpn-radiusplugin/archive/refs/heads/master.zip
+ unzip master.zip
+ cd openvpn-radiusplugin-master
+
+ Then apply this patch:
```
Comment (JeGoi) [inline docs/network/vpn_configuration.asciidoc]:
```suggestion
Then apply this patch:
```

File: lib/pf/Switch/OpenVPN.pm [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Redefinition of pf::Switch::parseVPNRequest due to specific attribute being used
+
+=cut
+
+sub parseVPNRequest {
+    my ( $self, $radius_request ) = @_;
+    my $logger = $self->logger;
+
+    my $client_ip       = $radius_request->{'Calling-Station-Id'};
+    my $mac             = '02:00:' . join(':', map { sprintf("%02x", $_) } split /\./, $radius_request->{'Calling-Station-Id'});
```
Comment (jrouzierinverse) [inline lib/pf/Switch/OpenVPN.pm:125]: Is the Calling-Station-Id always an IPv4 address?

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Review by JeGoi — COMMENTED** [review]
Need to change indent in doc

File: lib/pf/Switch/OpenVPN.pm (hunk 3 above)
Comment (fdurand) [reply lib/pf/Switch/OpenVPN.pm:125]: In the case of OpenVPN , yes

**Review by JeGoi — APPROVED** [review]

**Comment by fdurand** [comment]
BUMP

**Pull request merged**